src/types/member.ts

```diff
@@ -97,6 +97,10 @@ export type MemberAdminProps = {
 
   consumption?: number
   coins?: number
+  permissionGroups?: {
+    id: string
+    name: string
+  }[]
```

Comment:
NITS: can compose `MemberAdminProps` and `PermissionGroupProps` types at `useMemberAdmin` hook

Reply:
```ts
type Member = {
   id: string
   name: string
}
type MemberPhone = {
   memberId: string
   phone: string
}
type MemberPermission = {
   memberId: string
   permission: Permission
}
type Permission = {
   id: string
}

type MemberCardProps = {
    member: Member & {
        memberPhones: MemberPhones[]
    }
}

/********************/

type Member = {
   id: string
   name: string
   memberPhones: {
       memberId: string
       phone: string
   }[]
   memberPermissions: {
       memberId: string
       permission: Permission
   }[]
}

type MemberCardProps = {
    member: Omit<Member, 'memberPermissions'>
}

/********************/

const MemberCard<MemberCardProps> = ({ member }) => {}

```

Reply:
```ts
const member: Pick<Member, 'id' | 'name' | 'permissionIds' | ...> = {
    id: data.member_by_pk.id,
    name: data.member_by_pk.id,
    permissionIds: data.member_by_pk.member_permissions.map(v => v.permissionId),
}

```